includes/functions.php

```diff
@@ -1547,3 +1549,48 @@ function is_self_ping( $id ) {
 
 	return false;
 }
+
+/**
+ * Add an object to the outbox.
+ *
+ * @param mixed   $data               The object to add to the outbox.
+ * @param string  $type               The type of the Activity.
+ * @param integer $user_id            The User-ID.
+ * @param string  $content_visibility The visibility of the content.
+ *
+ * @return boolean|int The ID of the outbox item or false on failure.
+ */
+function add_to_outbox( $data, $type = 'Create', $user_id = 0, $content_visibility = ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC ) {
+	$transformer = Transformer_Factory::get_transformer( $data );
+
+	if ( ! $transformer || is_wp_error( $transformer ) ) {
+		return false;
+	}
+
+	$activity = $transformer->to_object();
+
+	if ( ! $activity || is_wp_error( $activity ) ) {
+		return false;
+	}
+
+	set_wp_object_state( $data, 'federate' );
+
+	$id = Outbox::add( $activity, $type, $user_id, $content_visibility );
+
+	if ( ! $id ) {
+		return false;
+	}
+
+	$hook = 'activitypub_process_outbox';
+	$args = array( $id );
+
+	if ( false === wp_next_scheduled( $hook, $args ) ) {
+		\wp_schedule_single_event(
+			\time() + 10,
+			$hook,
+			$args
+		);
+	}
+
```

Comment:
```suggestion

	if ( 'Create' === $type && ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC && $content_visibility && ACTIVITYPUB_ACTOR_AND_BLOG_MODE === \get_option( 'activitypub_actor_mode', ACTIVITYPUB_ACTOR_MODE ) ) {
		// @todo transform $data to announce
		add_to_outbox( function add_to_outbox( $data, 'Announce', Actors::BLOG_USER_ID );
	}
```

Reply:
you only share the id of a post via an `Announce`, so only `Creates` are important for that... the updates will be done automatically afterwards... no need to re-announce them...

Reply:
> you only share the id of a post via an Announce, so only Creates are important

Perfect. But the further question:

Let's say I follow a WP `example.com` site from `mattwiebe@mastodon.social` with `ACTIVITYPUB_ACTOR_AND_BLOG_MODE === true`. I follow `example.com@example.com`, but I don't follow `user@example.com`. I receive the `Announce` of a post by `user@example.com`. But the user then edits the post. Will `mastodon.social` get notified of the edit with nobody on the instance following `user@example.com`?

Reply:
🤔 good point...

Reply:
So this means that we'll need to also POST to the inboxes of the Blog Actor for`Update`s. Are there any other actions you can think of that should be included in that behaviour?

I'll add a commit to #1186 to do that.

Reply:
I would start with these two

Reply:
(and I am still not sure that update needs to be federated)

Reply:
the current code only supports `Create`, `Update` and `Delete` https://github.com/Automattic/wordpress-activitypub/blob/1f7abb76ef4fa171ec7d120eaa03eac3913d3c78/includes/class-activity-dispatcher.php#L104

Reply:
Perfect. I did `Update` but I added `Delete` as well - https://github.com/Automattic/wordpress-activitypub/pull/1186/commits/2caab9bdd6f7488f944fce5b6a132b0413f298d8